registration/api/v2/views.py

```diff
@@ -235,3 +238,92 @@ def get_queryset(self):
             return BetaTestApplication.objects.filter()
         else:
             return BetaTestApplication.objects.filter(user=self.request.user)
+
+
+class OpenEdxInstanceDeploymentViewSet(RetrieveDestroyAPIView, GenericViewSet):
+    """
+    Open edX Instance Deployment API.
+
+    This API can be used to manage the configuration for Open edX instances
+    owned by clients.
+    """
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        """
+        Get `BetaTestApplication` instances owned by current user.
+        For a regular user it should return a single object (for now).
+        """
+        user: User = self.request.user
+        if not user.is_authenticated:
+            return BetaTestApplication.objects.none()
+        elif user.is_staff:
+            return BetaTestApplication.objects.filter()
+        else:
+            return BetaTestApplication.objects.filter(user=self.request.user)
+
+    def get_serializer_class(self):
+        if self.action == 'delete':
+            return {}
+
+        return OpenEdXInstanceDeploymentSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        application = self.get_object()
+        instance = application.instance
+        deployment_status = constants.NO_STATUS
+
+        # Get latest active appserver
+        if not instance or not instance.get_active_appservers().exists():
+            deployment_status = constants.PREPARING_INSTANCE
+            num_changes = 0
+        else:
+            deployment_status = constants.UP_TO_DATE
+
+            # Get number of undeployed changes
+            num_changes = 0
+            if instance:
+                num_changes += 1 if application.instance_name != instance.name else 0
+                num_changes += 1 if application.privacy_policy_url != instance.privacy_policy_url else 0
+                num_changes += 1 if application.public_contact_email != instance.email else 0
+                if application.use_advanced_theme:
+                    # TODO: Improve counting differences of theming changes
+                    num_changes += 1 if application.draft_theme_config != instance.theme_config else 0
```

Comment:
I think this structure is a bit confusing. I understand it avoids multiple levels of ifs, however the else 0 looks a bit messy. Could you just switch these to standard ifs without an else? For example:
```suggestion
                    if application.draft_theme_config != instance.theme_config: num_changes += 1 
```

Reply:
Another approach is:

```python
num_changes = sum((
    application.instance_name != instance.name,
    application.privacy_policy_url != instance.privacy_policy_url,
    application.public_contact_email != instance.email,
    application.use_advanced_theme & application.draft_theme_config != instance.theme_config,
))
```